tests/cephadm/test_custom_container.py

```diff
@@ -0,0 +1,289 @@
+    config_ = get_cephci_config()
+    username_ = config_.get("cdn_credentials", {}).get("username")
+    password_ = config_.get("cdn_credentials", {}).get("password")
+
+    # Adding node_exporter,grafana_container_image,
+    # prometheus_container_image,alertmanager_container_image
+    for img in image:
+        img_reg = img.replace("registry.redhat.io/", "")
+        ceph.exec_command(
+            cmd=f"sudo podman run -v {OPT_CERTS}:/certs:Z \
+                        -v {OPT_CERTS}/domain.cert:/certs/domain.cert:Z \
+                        --rm registry.redhat.io/rhel8/skopeo:8.5-8 skopeo copy  \
+                        --remove-signatures --src-creds {username_}:{password_} \
+                        --dest-cert-dir=./certs/ --dest-creds \
+                        {PRIVATE_REG_USER}:{PRIVATE_REG_PASS} \
+                        docker://{img} \
+                        docker://{ceph.hostname}:5000/{img_reg} --tls-verify=false",
+            timeout=1200,
+        )
+
+    # List images in private registry
+    ceph.exec_command(
+        cmd=f"curl -u {PRIVATE_REG_USER}:{PRIVATE_REG_PASS} \
+                https://{ceph.hostname}:5000/v2/_catalog"
+    )
+
+
+def run(ceph_cluster, **kw):
+    """Verify re-deploying of monitoring stack with custom images
+    Args:
+        ceph_cluster(ceph.ceph.Ceph): CephNode or list of CephNode object
+    """
+
+    # Get installer node
+    node = ceph_cluster.get_nodes(role="installer")[0]
+
+    # Get config file and node
+    config = kw.get("config")
+    ceph_nodes = kw.get("ceph_nodes")
+
+    # Install httpsd-tools and httpd
+    node.exec_command(
+        sudo=True,
+        cmd="dnf install -y httpd-tools httpd",
+        long_running=True,
+    )
+
+    # Create folders for registry
+    node.exec_command(sudo=True, cmd="mkdir -p /opt/registry/{auth,certs,data}")
+
+    # Create credentials for accessing the private registry
+    node.exec_command(
+        sudo=True,
+        cmd=f"htpasswd -bBc {OPT_AUTH}/htpasswd \
+             {PRIVATE_REG_USER} {PRIVATE_REG_PASS}",
+    )
+
+    # Create a self-signed certificate
+    node.exec_command(
+        cmd=f'(echo "."; echo ".";echo ".";echo ".";echo ".";\
+                echo "{node.hostname}";echo ".";)| sudo openssl req -newkey \
+                rsa:4096 -nodes -sha256 -keyout {OPT_CERTS}/domain.key \
+                -x509 -days 365 -out {OPT_CERTS}/domain.crt -addext \
+                "subjectAltName = DNS:{node.hostname}"'
+    )
+
+    # Create a symbolic link to domain.cert
+    node.exec_command(
+        sudo=True,
+        cmd=f"ln -s {OPT_CERTS}/domain.crt \
+            {OPT_CERTS}/domain.cert",
+    )
+
+    # Add certificate to trusted list on the private registry node
+    node.exec_command(
+        sudo=True,
+        cmd=f"cp {OPT_CERTS}/domain.crt \
+            /etc/pki/ca-trust/source/anchors/",
+    )
+    node.exec_command(sudo=True, cmd="update-ca-trust")
+    update_cert(node)
```

Comment:
Shouldn't we need a check to verify whether the cert update is success or not? As per the changes suggested by Vaibhav, return of 0 and 1 needs to be added. Based on that we can do a check if the process was sucess or not